cpu/cc2538/periph/gpio.c

```diff
@@ -26,19 +26,27 @@
 #include "periph/gpio.h"
 
 #define GPIO_MASK           (0xfffff000)
-#define PORTNUM_MASK        (0x00007000)
+#define PORTNUM_MASK        (0x00003000)
```

Comment:
why changing this, I think `0x00007000` is right because `PORT_D = 3 = b11` and base address of `GPIO_A = 0x400d9000 = b1101 1001 0000 0000` hence
```
  b0000 0011 0000 0000 (=PORT_D << 12)
+ b1101 1001 0000 0000 (=GPIO_A)
= b1101 1100 0000 0000
          ^^ -> 0x00003000 (PORTNUM_MASK)
```
but with the old
```
= b1101 1100 0000 0000
         ^^^ -> 0x00007000 (PORTNUM_MASK)
```

the new portnum mask would not capture port_d, or does it?